File: Data Structures & Algorithms/daily-temperatures/submission-2.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        vector<int> res(temperatures.size(),0);
        stack<pair<int, int>> st;

        for (int i=0; i<temperatures.size(); i++) {
            int temperature = temperatures[i];
            while (!st.empty() && st.top().first < temperature) {
                pair<int,int> temp = st.top(); st.pop();
                res[temp.second] = i - temp.second;
            }
            st.push({temperature, i});
        }
        return res;
    }
};

File: Data Structures & Algorithms/validate-parentheses/submission-2.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        dictionary = {'(':')', '[':']', '{':'}'}
        for p in s:
            if p in dictionary.keys():
                stack.append(p)
            elif stack and dictionary[stack[-1]] == p:
                stack.pop()
            elif (stack and dictionary[stack[-1]] != p) or (not stack and p not in dictionary.keys()):
                return False
        return True if len(stack) == 0 else False